misc/test_runner/include/ia2_test_runner.h

```diff
@@ -17,25 +17,40 @@ typedef void (*ia2_test_fn)(void);
 #endif
 
 struct fake_criterion_test {
+  struct fake_criterion_test *next;
+  const char *suite;
+  const char *name;
   ia2_test_fn test;
   ia2_test_fn init;
   int exit_code;
 };
 
+extern struct fake_criterion_test *fake_criterion_tests;
+
+#define _STRINGIFY(a) #a
+#define STRINGIFY(a) _STRINGIFY(a)
+
 /*
  * Placing IA2_{BEGIN,END}_NO_WRAP between the function declaration stops the rewriter from creating a
  * direct call gate for the test function or indirect call gates for function pointer expressions
  * that reference it like the RHS when initializing struct fake_criterion_test's test field. The
  * last line of this macro is the start of the test function's definition and should be followed by { }
  */
-#define Test(suite, name, ...)                                                                                                 \
-  IA2_BEGIN_NO_WRAP                                                                                                            \
-  void fake_criterion_##suite##_##name(void);                                                                                  \
-  IA2_END_NO_WRAP                                                                                                              \
-  __attribute__((__section__("fake_criterion_tests"))) struct fake_criterion_test fake_criterion_##suite##_##name##_##test = { \
-      .test = fake_criterion_##suite##_##name,                                                                                 \
-      ##__VA_ARGS__};                                                                                                          \
-  void fake_criterion_##suite##_##name(void)
+#define Test(suite_, name_, ...)                                                           \
+  IA2_BEGIN_NO_WRAP                                                                        \
+  void fake_criterion_##suite_##_##name_(void);                                            \
+  IA2_END_NO_WRAP                                                                          \
+  struct fake_criterion_test fake_criterion_##suite_##_##name_##_##test = {                \
+      .next = NULL,                                                                        \
+      .suite = STRINGIFY(suite_),                                                          \
+      .name = STRINGIFY(name_),                                                            \
+      .test = fake_criterion_##suite_##_##name_,                                           \
+      ##__VA_ARGS__};                                                                      \
```

Comment:
I think these structs should have `IA2_SHARED_DATA` to prevent potential test flakiness.

Reply:
Fixed in [e9367d6..b32c3ce](https://github.com/immunant/IA2-Phase2/compare/e9367d68ce41a2f8af4370183efe3132f71dd46f..b32c3ce63e7df8569847173526285ed916ecc1f6).